src/types/transaction.ts

```diff
@@ -151,6 +153,42 @@ export function toTypedEthTx({
   }
 }
 
+/**
+ * @param bytes - The bytes of the rlp encoded transaction without signature.
+ * For Legacy = rlp([nonce, gasprice, startgas, to, value, data, chainid, 0, 0])
+ * For EIP1559 = [0x02 || rlp([chain_id, nonce, max_priority_fee_per_gas, max_fee_per_gas, gas_limit, destination, amount, data, access_list])]
+ * For EIP2930 = [0x01 || rlp([chainId, nonce, gasPrice, gasLimit, to, value, data, accessList])]
+ * @returns - Decoded unsigned transaction.
+ * @throws - Error if the transaction is a BlobEIP4844Tx or the rlp encoding is not an array.
+ */
+function fromSerializedData(
+  bytes: Uint8Array,
+): TypedTransaction {
+  const txType = bytes[0];
+  if (txType <= 0x7f) {
+    switch (txType) {
+      case TransactionType.AccessListEIP2930:
+        return AccessListEIP2930Transaction.fromSerializedTx(bytes);
+      case TransactionType.FeeMarketEIP1559:
+        return FeeMarketEIP1559Transaction.fromSerializedTx(bytes);
+      default:
+        throw new Error(`Invalid tx type: ${txType}`);
+    }
+  } else {
+    const values = RLP.decode(bytes);
+    if (!Array.isArray(values)) {
+      throw new Error("Invalid serialized tx input: must be array");
+    }
+    const legacyTxValues = values as TxValuesArray[TransactionType.Legacy];
+    // In the case of a Legacy, we need to update the chain id to be a value >= 37.
+    const chainId = bytesToBigInt(legacyTxValues[6]);
+    values[6] = chainId > 37n ? values[6] : bigIntToBytes(37n);
```

Comment:
I still don't understand the need for this

Reply:
in order to deserialize the transactions, we can use `fromSerializedTx`, which calls `RLP.decode` and then `fromValuesArray`. In the case of Fee Market and EIP2930 transactions, no issue. Now for Legacy transactions, this will decode to `[nonce, gasprice, startgas, to, value, data, chainid, 0, 0]` and typescript will assume that you are using v = chainId, r and s = 0. This will fail if chainId is lower than 37 and throw.

Reply:
But why does JS assume that v = chainId, r = s = 0?